Index: src/org/thechiselgroup/choosel/public/js/lib/timeline-2.3.0/scripts/overview-painter.js
===================================================================
--- src/org/thechiselgroup/choosel/public/js/lib/timeline-2.3.0/scripts/overview-painter.js	(revision 281)
+++ src/org/thechiselgroup/choosel/public/js/lib/timeline-2.3.0/scripts/overview-painter.js	(working copy)
@@ -219,6 +219,12 @@
     tickDiv.style.top = top + "px";
     
     tickDiv.id = this._encodeEventElID('tick', evt);
+    if (evt._tickZIndex) {
+    	tickDiv.style.zIndex = evt._tickZIndex;
+    }
+    if (evt._tickBackgroundColor) {
+    	tickDiv.style.backgroundColor = evt._tickBackgroundColor;
+    }
   //  tickDiv.style.width = width + "px";
   //  tickDiv.style.position = "absolute";
   //  tickDiv.style.height = height + "px";
